pkg/controller/podautoscaler/horizontal.go

```diff
@@ -520,6 +486,57 @@ func (a *HorizontalController) reconcileAutoscaler(hpav1Shared *autoscalingv1.Ho
 	return a.updateStatusIfNeeded(hpaStatusOriginal, hpa)
 }
 
+// normalizeDesiredReplicas takes the metrics desired replicas value and normalizes it based on the appropriate conditions (i.e. < maxReplicas, >
+// minReplicas, etc...)
+func (a *HorizontalController) normalizeDesiredReplicas(hpa *autoscalingv2.HorizontalPodAutoscaler, currentReplicas int32, prenormalizedDesiredReplicas int32) (int32, error) {
+	var minReplicas int32
+	if hpa.Spec.MinReplicas != nil {
+		minReplicas = *hpa.Spec.MinReplicas
+	} else {
+		minReplicas = 0
+	}
+
+	desiredReplicas, err := convertDesiredReplicasWithRules(currentReplicas, prenormalizedDesiredReplicas, minReplicas, hpa.Spec.MaxReplicas)
+	if err != nil {
+		return -1, err
+	}
+
+	if desiredReplicas < prenormalizedDesiredReplicas {
+		setCondition(hpa, autoscalingv2.ScalingLimited, v1.ConditionTrue, "TooManyReplicas", "the desired replica count is above the maximum number of allowed replicas")
```

Comment:
you've lost `ScaleUpLimit` here, which I think is a useful condition.  We should find a way to restore it.

Reply:
Yes good point! I initially couldn't thing of a way to preserve `ScaleUpLimit` without passing around the `hpa` struct - which I was hesitant to do for testing reasons. But I was thinking about it last night, and think I thought of an option.